src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/CreateIndexAttempt.java

```diff
@@ -0,0 +1,143 @@
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+ An attempt to create index for a table's column.
+*/
+public class CreateIndexAttempt extends SchemaAttempt<TableSchemaObject> {
+  private final String columnName;
+  private final String indexName;
+  private final TextIndexOptions textIndexOptions;
+  private final VectorIndexOptions vectorIndexOptions;
+  private final DataType dataType;
+  private final boolean ifNotExists;
+
+  /*
+   * @param position The position of the attempt in the sequence, for create index it's  always 0.
+   * @param schemaObject The schema object representing the table.
+   * @param columnName The name of the column to create the index on.
+   * @param dataType The data type of the column.
+   * @param indexName The name of the index to create.
+   * @param textIndexOptions The options for a text column type index.
+   * @param vectorIndexOptions The options for a vector column type index.
+   * @param ifNotExists Flag to ignore if index already exists.
+   * @return The attempt to create the index.
+   */
+  protected CreateIndexAttempt(
+      int position,
+      TableSchemaObject schemaObject,
+      String columnName,
+      DataType dataType,
+      String indexName,
+      TextIndexOptions textIndexOptions,
+      VectorIndexOptions vectorIndexOptions,
+      boolean ifNotExists) {
+    super(position, schemaObject, new SchemaRetryPolicy(2, Duration.ofMillis(10)));
+
+    this.columnName = columnName;
+    this.dataType = dataType;
+    this.indexName = indexName;
+    this.textIndexOptions = textIndexOptions;
+    this.vectorIndexOptions = vectorIndexOptions;
+    this.ifNotExists = ifNotExists;
+    setStatus(OperationStatus.READY);
+  }
+
+  /*
+   * Options for a text index.
+   */
+  public record TextIndexOptions(Boolean caseSensitive, Boolean normalize, Boolean ascii) {
+
+    public Map<String, Object> getOptions() {
+      Map<String, Object> options = new HashMap<>();
+      if (caseSensitive != null) {
+        options.put("case_sensitive", caseSensitive);
+      }
+      if (normalize != null) {
+        options.put("normalize", normalize);
+      }
+      if (ascii != null) {
+        options.put("ascii", ascii);
+      }
+      return options;
+    }
+  }
+
+  /*
+   * Options for a vector index.
+   */
+  public record VectorIndexOptions(SimilarityFunction similarityFunction, String sourceModel) {
+    public Map<String, Object> getOptions() {
+      Map<String, Object> options = new HashMap<>();
+      if (similarityFunction != null) {
+        options.put("similarity_function", similarityFunction.getMetric());
+      }
+      if (sourceModel != null) {
+        options.put("source_model", sourceModel);
+      }
+      return options;
+    }
+  }
+
+  @Override
+  protected SimpleStatement buildStatement() {
+    var keyspaceIdentifier = cqlIdentifierFromUserInput(schemaObject.name().keyspace());
```

Comment:
In this case, use of actual type (`CqlIdentifier`? or `String`?) would seem more readable to me than `var` (even if bit longer)